### PR TITLE
Using JSON for script arguments

### LINKED ISSUE #67: Improve the way scripts take arguments
To avoid having the issue of sending many arguments to the scripts that are then retrieved in a very specific order, send a JSON string in a single argument that the scripts can easily parse and read what they need from.

### PR DESCRIPTION
**Pull request by Cerfoglg**
Fixes #67
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/benchflow/data-transformers/pull/66%23discussion_r64745331%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-221876481%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-221959293%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-222106973%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-222108125%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-222113495%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23discussion_r64932529%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23discussion_r65107205%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-222557292%22%2C%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-222622917%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23issuecomment-221876481%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%5Cn%21%5BCodacy%5D%28https%3A//www.codacy.com/assets/images/favicon.png%29%20Here%20is%20an%20overview%20of%20what%20got%20changed%20by%20this%20pull%20request%3A%5Cn%5Cn%5Cn%60%60%60diff%5Cn%5CnIssues%5Cn%3D%3D%3D%3D%3D%3D%5Cn%2B%20Solved%201%5Cn-%20Added%201%5Cn%20%20%20%20%20%20%20%20%20%20%20%5Cn%60%60%60%5Cn%5Cn%5CnSee%20the%20complete%20overview%20on%20%5BCodacy%5D%28https%3A//www.codacy.com/app/BenchFlow/data-transformers/pullRequest%3Fprid%3D244584%26bid%3D3321021%29%5Cn%20%20%20%20%20%20%20%20%22%2C%20%22created_at%22%3A%20%222016-05-26T13%3A51%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22Map%20the%20issues%20as%20well%2C%20as%20usual%22%2C%20%22created_at%22%3A%20%222016-05-26T18%3A45%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%5Cn%21%5BCodacy%5D%28https%3A//www.codacy.com/assets/images/favicon.png%29%20Here%20is%20an%20overview%20of%20what%20got%20changed%20by%20this%20pull%20request%3A%5Cn%5Cn%5Cn%60%60%60diff%5Cn%5CnIssues%5Cn%3D%3D%3D%3D%3D%3D%5Cn%2B%20Solved%201%5Cn-%20Added%202%5Cn%20%20%20%20%20%20%20%20%20%20%20%5Cn%60%60%60%5Cn%5Cn%5CnSee%20the%20complete%20overview%20on%20%5BCodacy%5D%28https%3A//www.codacy.com/app/BenchFlow/data-transformers/pullRequest%3Fprid%3D244584%26bid%3D3321021%29%5Cn%20%20%20%20%20%20%20%20%22%2C%20%22created_at%22%3A%20%222016-05-27T09%3A52%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20fixes%20%2367%22%2C%20%22created_at%22%3A%20%222016-05-27T09%3A58%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40Cerfoglg%20fine%2C%20just%20solve%20the%20remaining%20issues%20reported%20by%20Codacy%22%2C%20%22created_at%22%3A%20%222016-05-27T10%3A27%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40Cerfoglg%20remaining%3A%20https%3A//www.codacy.com/app/BenchFlow/data-transformers/pullRequest%3Fprid%3D244584%22%2C%20%22created_at%22%3A%20%222016-05-30T21%3A36%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40VincenzoFerme%20Done%22%2C%20%22created_at%22%3A%20%222016-05-31T08%3A22%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8819678%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Cerfoglg%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20737b546e65bfeb0a0b10333fabed5cc10456eca4%20data-transformers/transformers/propertiesTransformer.py%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23discussion_r64932529%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%21%5BCodacy%5D%28https%3A//www.codacy.com/assets/images/favicon.png%29%20Issue%20found%3A%20%5BUnused%20variable%20%27SUTName%27%5D%28https%3A//www.codacy.com/app/BenchFlow/data-transformers/file/2647790838/issues/source%3Fbid%3D3301038%26fileBranchId%3D3321021%23l231%29%22%2C%20%22created_at%22%3A%20%222016-05-27T16%3A36%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20data-transformers/transformers/propertiesTransformer.py%3AL223-238%22%7D%2C%20%22Pull%207a304fd6089a5c7f43bc144b65dfed250127db2d%20data-transformers/transformers/statsTransformer.py%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/benchflow/data-transformers/pull/66%23discussion_r64745331%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%21%5BCodacy%5D%28https%3A//www.codacy.com/assets/images/favicon.png%29%20Issue%20found%3A%20%5BReimport%20%27json%27%20%28imported%20line%202%29%5D%28https%3A//www.codacy.com/app/BenchFlow/data-transformers/file/2639145825/issues/source%3Fbid%3D3301038%26fileBranchId%3D3321021%23l8%29%22%2C%20%22created_at%22%3A%20%222016-05-26T13%3A34%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-05-30T21%3A36%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7163641%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VincenzoFerme%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20data-transformers/transformers/statsTransformer.py%3AL5-12%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/66#issuecomment-221876481'>General Comment</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Codacy Here is an overview of what got changed by this pull request:

``` diff
Issues
======
+ Solved 1
- Added 1
```

See the complete overview on [Codacy](https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584&bid=3321021)
- <a href='https://github.com/VincenzoFerme'></a> Map the issues as well, as usual
- <a href='https://github.com/VincenzoFerme'></a> Codacy Here is an overview of what got changed by this pull request:

``` diff
Issues
======
+ Solved 1
- Added 2
```

See the complete overview on [Codacy](https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584&bid=3321021)
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme fixes #67
- <a href='https://github.com/VincenzoFerme'></a> @Cerfoglg fine, just solve the remaining issues reported by Codacy
- <a href='https://github.com/VincenzoFerme'></a> @Cerfoglg remaining: https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584
- <a href='https://github.com/Cerfoglg'></a> @VincenzoFerme Done
- [x] <a href='#crh-comment-Pull 737b546e65bfeb0a0b10333fabed5cc10456eca4 data-transformers/transformers/propertiesTransformer.py 16'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/66#discussion_r64932529'>File: data-transformers/transformers/propertiesTransformer.py:L223-238</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Codacy Issue found: [Unused variable 'SUTName'](https://www.codacy.com/app/BenchFlow/data-transformers/file/2647790838/issues/source?bid=3301038&fileBranchId=3321021#l231)
- [x] <a href='#crh-comment-Pull 7a304fd6089a5c7f43bc144b65dfed250127db2d data-transformers/transformers/statsTransformer.py 4'></a> &nbsp;<b><a href='https://github.com/benchflow/data-transformers/pull/66#discussion_r64745331'>File: data-transformers/transformers/statsTransformer.py:L5-12</a></b>
- <a href='https://github.com/VincenzoFerme'></a> Codacy Issue found: [Reimport 'json' (imported line 2)](https://www.codacy.com/app/BenchFlow/data-transformers/file/2639145825/issues/source?bid=3301038&fileBranchId=3321021#l8)

<a href='https://www.codereviewhub.com/benchflow/data-transformers/pull/66?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/benchflow/data-transformers/pull/66?approve=1'></a>&nbsp;<a href='https://github.com/benchflow/data-transformers/pull/66'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by VincenzoFerme** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/benchflow/data-transformers/pull/66'>Refresh</a>.<a href='#crh-update-none'></a>

File: data-transformers/transformers/statsTransformer.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import gzip
 import uuid
 import yaml
+import json
```
Comment (VincenzoFerme) [inline data-transformers/transformers/statsTransformer.py]: Codacy Issue found: [Reimport 'json' (imported line 2)](https://www.codacy.com/app/BenchFlow/data-transformers/file/2639145825/issues/source?bid=3301038&fileBranchId=3321021#l8)

File: data-transformers/transformers/propertiesTransformer.py [hunk 2]
```diff
@@ -223,13 +223,15 @@ def main():
...
-    experimentID = sys.argv[4]
-    SUTName = sys.argv[5]
-    containerID = sys.argv[6]
-    hostID = sys.argv[7]
+    args = json.loads(sys.argv[1])
+    minioHost = args["minio_host"]
+    filePath = args["file_path"]
+    trialID = args["trial_id"]
+    experimentID = args["experiment_id"]
+    SUTName = args["sut_name"]
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py]: Codacy Issue found: [Unused variable 'SUTName'](https://www.codacy.com/app/BenchFlow/data-transformers/file/2639145824/issues/source?bid=3301038&fileBranchId=3321021#l231)

**Comment by VincenzoFerme** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584&bid=3321021)

File: data-transformers/transformers/propertiesTransformer.py (hunk 2 above)
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py]: Codacy Issue found: [Unused variable 'SUTName'](https://www.codacy.com/app/BenchFlow/data-transformers/file/2639285900/issues/source?bid=3301038&fileBranchId=3321021#l231)

**Comment by VincenzoFerme** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584&bid=3321021)

**Comment by VincenzoFerme** [comment]
Map the issues as well, as usual

File: data-transformers/transformers/propertiesTransformer.py [hunk 3]
```diff
@@ -223,13 +223,15 @@ def main():
...
-    experimentID = sys.argv[4]
-    SUTName = sys.argv[5]
-    containerID = sys.argv[6]
-    hostID = sys.argv[7]
+    args = json.loads(sys.argv[1])
+    minioHost = str(args["minio_host"])
+    filePath = str(args["file_path"])
+    trialID = str(args["trial_id"])
+    experimentID = str(args["experiment_id"])
+    SUTName = str(args["sut_name"])
```
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py]: Codacy Issue found: [Unused variable 'SUTName'](https://www.codacy.com/app/BenchFlow/data-transformers/file/2644900856/issues/source?bid=3301038&fileBranchId=3321021#l231)

File: data-transformers/transformers/statsTransformer.py [hunk 4]
```diff
@@ -107,13 +107,15 @@ def main():
...
-    experimentID = sys.argv[4]
-    #SUTName = sys.argv[5]
-    containerID = sys.argv[6]
-    hostID = sys.argv[7]
+    args = json.loads(sys.argv[1])
+    minioHost = str(args["minio_host"])
+    filePath = str(args["file_path"])
+    trialID = str(args["trial_id"])
+    experimentID = str(args["experiment_id"])
+    SUTName = str(args["sut_name"])
```
Comment (VincenzoFerme) [inline data-transformers/transformers/statsTransformer.py]: Codacy Issue found: [Unused variable 'SUTName'](https://www.codacy.com/app/BenchFlow/data-transformers/file/2644900857/issues/source?bid=3301038&fileBranchId=3321021#l115)

**Comment by VincenzoFerme** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584&bid=3321021)

**Comment by Cerfoglg** [comment]
@VincenzoFerme fixes #67

**Comment by VincenzoFerme** [comment]
@Cerfoglg fine, just solve the remaining issues reported by Codacy

File: data-transformers/transformers/statsTransformer.py (hunk 4 above)
Comment (VincenzoFerme) [inline data-transformers/transformers/statsTransformer.py]: Codacy Issue found: [Unused variable 'SUTName'](https://www.codacy.com/app/BenchFlow/data-transformers/file/2647790839/issues/source?bid=3301038&fileBranchId=3321021#l115)

File: data-transformers/transformers/propertiesTransformer.py (hunk 3 above)
Comment (VincenzoFerme) [inline data-transformers/transformers/propertiesTransformer.py]: Codacy Issue found: [Unused variable 'SUTName'](https://www.codacy.com/app/BenchFlow/data-transformers/file/2647790838/issues/source?bid=3301038&fileBranchId=3321021#l231)

**Comment by VincenzoFerme** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584&bid=3321021)

File: data-transformers/transformers/statsTransformer.py (hunk 1 above)
Comment (VincenzoFerme) [inline data-transformers/transformers/statsTransformer.py]: :+1:

**Comment by VincenzoFerme** [comment]
@Cerfoglg remaining: https://www.codacy.com/app/BenchFlow/data-transformers/pullRequest?prid=244584

**Comment by Cerfoglg** [comment]
@VincenzoFerme Done

**Pull request merged**